pkg/users/client.go

```diff
@@ -0,0 +1,183 @@
+package users
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/workos/workos-go/v2/internal/workos"
+	"github.com/workos/workos-go/v2/pkg/workos_errors"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+// ResponseLimit is the default number of records to limit a response to.
+const ResponseLimit = 10
+
+// Order represents the order of records.
+type Order string
+
+// Constants that enumerate the available orders.
+const (
+	Asc  Order = "asc"
+	Desc Order = "desc"
+)
+
+// Client represents a client that fetch SSO data from WorkOS API.
+type Client struct {
+	// The WorkOS api key. It can be found in
+	// https://dashboard.workos.com/api-keys.
+	//
+	// REQUIRED.
+	APIKey string
+
+	// The WorkOS Client ID (eg. client_01JG3BCPTRTSTTWQR4VSHXGWCQ).
+	//
+	// REQUIRED.
+	ClientID string
+
+	// The endpoint to WorkOS API.
+	//
+	// Defaults to https://api.workos.com.
+	Endpoint string
+
+	// The http.Client that is used to send request to WorkOS.
+	//
+	// Defaults to http.Client.
+	HTTPClient *http.Client
+
+	// The function used to encode in JSON. Defaults to json.Marshal.
+	JSONEncode func(v interface{}) ([]byte, error)
+
+	once sync.Once
+}
+
+func (c *Client) init() {
```

Comment:
We should probably switch from this method to something more like the new SSO method. Instead of being set in client.go it can be set in users.go. 

A NewClient function is a more idiomatic approach. I'm not sure this is a case where lazy initialization has advantages enough to make the init() make more sense?






```
// NewClient returns a new instance of the Client struct with default values.
func NewClient(apiKey, clientID string) *Client {
	return &Client{
		APIKey:     apiKey,
		ClientID:   clientID,
		Endpoint:   "https://api.workos.com",
		HTTPClient: &http.Client{Timeout: time.Second * 10},
		JSONEncode: json.Marshal,
	}
}
```